activerecord/lib/active_record/connection_adapters/abstract_mysql_adapter.rb

```diff
@@ -11,6 +11,16 @@ def primary_key(name, type = :primary_key, options = {})
           options[:auto_increment] ||= type == :bigint
           super
         end
+
+        def new_column_definition(name, type, options) # :nodoc:
+          column = super
+          case column.type
+          when :primary_key
+            column.type = :integer
+            column.auto_increment = true
+          end
```

Comment:
Actually normalizing primary key as an integer is for respects `unsigned` option.
https://github.com/rails/rails/pull/17696/commits/f3772f729c72d098156b35eb105d3ffdd928c5eb#diff-98ce5dc0cd01f8b0435ff6feac10d8a8R33

If you want to create `bigserial` or `bigint` primary key for Postgres, simply you can specify it.

```ruby
    # bigserial
    create_table :foos, id: :bigserial do |t|
    end

    # bigint (without default sequence)
    create_table :foos, id: :bigint do |t|
    end
```